### PR TITLE
#96 - Modernization - Phase 1 - ES6 Modules - 1

### PR DESCRIPTION
**Pull request by parthopdas**
> Adding eslint + whitelisting only few files
> DPR protected global
> Moving entry point to ES6
> Moving features to ES6
> Adding & configuring husky

### CODE REVIEW

File: _dprhtml/js/dprviewmodel.js
Comment (catalin-i) [inline _dprhtml/js/dprviewmodel.js:66]: I think I'd prefer having the commands namespaced under the window object - think it would help with organization. I'm tempted to suggest a namespace for all the DPR globals - but that might be excessive (sacrificing succintness)

**Review by catalin-i — APPROVED** [review]
Looks good from my side. I suggested namespacing the DPR commands (since they are quite numerous) - but it's a matter of preference.

File: _dprhtml/js/dprviewmodel.js
Comment (parthopdas) [reply _dprhtml/js/dprviewmodel.js:66]: super - that is what I added the DPRGlobal class for. i also want to make most of the stuff write once considering it is a global variable.

i'll make that change before merging this commit

Comment (catalin-i) [reply _dprhtml/js/dprviewmodel.js:66]: Agree, since they're all defined in a single place, a global namespace seems redundant.

If you want to preserve the look and the succinctness, you can just convert the "DPD_CMD" prefix of the identifier into the namespace.

File: _dprhtml/js/index.js
Comment (rrogowski) [inline _dprhtml/js/index.js:58]: Can we remove this line? Looks like it was a first attempt that was replaced by the line below it.

File: .github/workflows/cd.yml
Comment (rrogowski) [inline .github/workflows/cd.yml:22]: Thanks for adding this 🙌

File: package.json
Comment (rrogowski) [inline package.json:34]: I might be in the minority here (and if so I'm happy to defer to majority opinion), but I find pre-commit/push hooks a bit annoying 😅. My editor lets me know when I am violating lint and/or style rules, and I run tests in watch mode while I develop. So in my case having extra checks run automatically is fairly redundant. That being said, if the total lint + test time took ~1s or less, it wouldn't really be a big deal, though that could change case as we add more tests.

**Review by rrogowski — APPROVED** [review]

**Comment by rrogowski** [comment]
@parthopdas Do we need both a `yarn.lock` and a `package-lock.json`? I think `package-lock.json` is generated via `npm install`, but I think it might be better to stick with just one. I'd advocate for `yarn.lock` since `yarn` downloads packages in parallel, and I believe `yarn.lock` creates fewer merge conflicts than `package-lock.json`.

File: package.json
Comment (parthopdas) [reply package.json:34]: yeah i do the same. 

the hooks i am think will be the 'training' for other that don't (including one off folks who contribute). in essence it will save them a push (and to an extent commit) round trip. this is esp. considering we've never had it in the code base so far.

File: _dprhtml/js/index.js
Comment (parthopdas) [reply _dprhtml/js/index.js:58]: my bad - nuked it.

**Comment by parthopdas** [comment]
@rrogowski no we don't - nuked the npm lock and added it to .gitignore - i think it was added by mistake.

File: package.json
Comment (rrogowski) [reply package.json:34]: @parthopdas sounds good to me, thanks for explaining!

File: _dprhtml/js/dprviewmodel.js
Comment (parthopdas) [reply _dprhtml/js/dprviewmodel.js]: i'll be doing this in the next commit as the size of this PR has increased quiet a bit due to infra stuff. let me know if that is no ok with you.

Comment (catalin-i) [reply _dprhtml/js/dprviewmodel.js]: Totally fine, no reason to have this PR too bloated.

**Pull request merged**
